extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -837,6 +925,32 @@ public static JsonArray getJsonResponse(@Nonnull final Page page,
                     .object("client")
                         .value("clientName", "ANDROID")
                         .value("clientVersion", MOBILE_YOUTUBE_CLIENT_VERSION)
+                        .value("platform", "MOBILE")
+                        .value("hl", localization.getLocalizationCode())
+                        .value("gl", contentCountry.getCountryCode())
+                    .end()
+                    .object("user")
+                        // TO DO: provide a way to enable restricted mode with:
+                        // .value("enableSafetyMode", boolean)
+                        .value("lockedSafetyMode", false)
+                    .end()
+                .end();
+        // @formatter:on
+    }
+
+    @Nonnull
+    public static JsonBuilder<JsonObject> prepareIosMobileJsonBuilder(
+            @Nonnull final Localization localization,
+            @Nonnull final ContentCountry contentCountry) {
+        // @formatter:off
+        return JsonObject.builder()
+                .object("context")
+                    .object("client")
+                        .value("clientName", "IOS")
```

Comment:
Shouldn't this value be in a constant?

Reply:
Probably not right now, but we should probably refactor how clients are managed in the extractor later, to deduplicate code used.